source/frontend/src/components/common/ExpandableTextList.tsx

```diff
@@ -25,12 +26,41 @@ export function ExpandableTextList<T>({
   renderFunction,
   delimiter,
   maxCollapsedLength,
+  displayedPerChunk,
   moreText,
   hideText,
 }: IExpandableTextListProps<T>) {
   const [isExpanded, setIsExpanded] = useState(false);
-  const displayedItemsLength = !isExpanded ? maxCollapsedLength ?? items.length : items.length;
+  const [displayedItemsLength, setDisplayedItemsLength] = useState(
+    !isExpanded ? maxCollapsedLength : displayedPerChunk ?? items.length,
+  );
```

Comment:
in the old logic `"maxCollapsedLength ?? items.length"` was protecting against maxCollapsedLength being **undefined/null**. In the new logic I don't see the same. Shouldn't we keep that logic in the new one?

Reply:
Yes, will make those changes